### ISSUE TITLE
`unused_parens` is not emitted when used in macro.

### ISSUE THREAD

**Issue by lcnr** [rust-lang/rust#70076]
```rust
fn main() {
    let x = vec![((8))];
    println!("Hello, world!, {:?}",x);
}
```
and
```rust
fn main() {
    println!("Hello, world!, {:?}", ((8)));
}
```
does not have any warning while
```rust
fn main() {
    let x = ((8));
    println!("Hello, world!, {:?}",x);
}
```

emits `unused_parens`.

I expect the first two snippets to also cause this warning.

**Comment by jonas-schievink** [comment]
> note: is there a way to add labels to an issue without being a member?

[Yes!](https://github.com/rust-lang/triagebot/wiki/Labeling)
